api/src/index.js

```diff
@@ -29,3 +19,86 @@ app.use("/api", apiRouter);
 app.listen(process.env.PORT, () => {
   console.log(`API listening on port ${process.env.PORT}`);
 });
+
+// This is a simple example of how to use Knex.js with Express.js
+app.get("/", (req, res) => {
+  const routes = [
+    { route: "/future-meals" },
+    { route: "/past-meals" },
+    { route: "/all-meals" },
+    { route: "/first-meal" },
+    { route: "/last-meal" },
+  ];
+
+  res.json({
+    message: "Hello! This is the MEAL SHARING APP",
+    routes: routes,
+  });
+});
+
+// ROUTE TO GET FUTURE MEALS. USED KNEX QUERY BUILDER
+app.get("/future-meals", async (req, res) => {
+  try {
+    const future_meals = await knex("meal")
+      .where("when", ">", new Date())
+      .orderBy("when", "asc");
+    res.json(future_meals);
+  } catch (error) {
+    console.error("Error fetching meals:", error);
+    res.status(500).json({ error: "Internal server error" });
+  }
+});
+
+// ROUTE TO GET PAST MEALS. USED KNEX QUERY BUILDER
+
+app.get("/past-meals", async (req, res) => {
+  try {
+    const past_meals = await knex("meal")
+      .where("when", "<", new Date())
+      .orderBy("when", "asc");
+    res.json(past_meals);
+  } catch (error) {
+    console.error("Error fetching meals:", error);
+    res.status(500).json({ error: "Internal server error" });
+  }
+});
+
+// ROUTE TO GET ALL MEALS SORTED BY ID. USING RAW SQL
+app.get("/all-meals", async (req, res) => {
+  try {
+    const result = await knex.raw("SELECT * FROM meal ORDER BY id ASC");
```

Comment:
minor: I notice you do not use knex query builder, but raw instead for the remaining endpoints, it is always good to be consistent i.e either knexJS or raw query :)